Rubberduck.Core/UI/ToDoItems/ToDoExplorerViewModel.cs

```diff
@@ -123,122 +119,109 @@ private void HandleStateChanged(object sender, EventArgs e)
+            _uiDispatcher.Invoke(() =>
+            {
+                _items.Clear();
+                foreach (var item in _state.AllComments.SelectMany(GetToDoMarkers))
+                {
+                    _items.Add(item);
+                }
+            });
         }
 
-        private ToDoItem _selectedItem;
-        public INavigateSource SelectedItem
+        public INavigateCommand NavigateCommand { get; }
+
+        public ReparseCommand RefreshCommand { get; set; }
+
+        public CommandBase RemoveCommand { get; }
+
+        public CommandBase CollapseAllCommand { get; }
+
+        public CommandBase ExpandAllCommand { get; }
+
+        public CommandBase CopyResultsCommand { get; }
+
+        public CommandBase OpenTodoSettingsCommand { get; }
+
+        private void ExecuteCollapseAll(object parameter)
         {
-            get => _selectedItem;
-            set
-            {
-                _selectedItem = value as ToDoItem; 
-                OnPropertyChanged();
-            }
+            ExpandedState = false;
         }
 
-        private CommandBase _removeCommand;
-        public CommandBase RemoveCommand
+        private void ExecuteExpandAll(object parameter)
         {
-            get
-            {
-                if (_removeCommand != null)
-                {
-                    return _removeCommand;
-                }
-                return _removeCommand = new DelegateCommand(LogManager.GetCurrentClassLogger(), _ =>
-                {
-                    if (_selectedItem == null)
-                    {
-                        return;
-                    }
+            ExpandedState = true;
+        }
 
-                    var component = _state.ProjectsProvider.Component(_selectedItem.Selection.QualifiedName);
-                    using (var module = component.CodeModule)
-                    {
-                        var oldContent = module.GetLines(_selectedItem.Selection.Selection.StartLine, 1);
-                        var newContent = oldContent.Remove(_selectedItem.Selection.Selection.StartColumn - 1);
+        private bool CanExecuteRemoveCommand(object obj) => SelectedItem != null && RefreshCommand.CanExecute(obj);
 
-                        module.ReplaceLine(_selectedItem.Selection.Selection.StartLine, newContent);
-                    }
+        private void ExecuteRemoveCommand(object obj)
+        {
+            if (!CanExecuteRemoveCommand(obj))
+            {
+                return;
+            }
 
-                    RefreshCommand.Execute(null);
-                }
-                );
+            var component = _state.ProjectsProvider.Component(_selectedItem.Selection.QualifiedName);
+            using (var module = component.CodeModule)
+            {
+                var oldContent = module.GetLines(_selectedItem.Selection.Selection.StartLine, 1);
+                var newContent = oldContent.Remove(_selectedItem.Selection.Selection.StartColumn - 1);
```

Comment:
This looks weird. We clear the matching line up to the start of the selection?

Reply:
The selection there is the comment (this also only shown in the diff because I moved it).  It's had that behavior as far back as I can go in the commit history.